### PR TITLE
Add a link content check before removing it from storage

### LINKED ISSUE #341: Failure when deleting an empty ScLink
```c++
ScAddr linkAddr = ctx.CreateLink();
ctx.EraseElement(linkAddr);
```
`sc-machine/sc-memory/sc-core/sc-store/sc_storage.c:481:sc_storage_element_free: assertion failed: (__r == SC_RESULT_OK)`

### PR DESCRIPTION
**Pull request by ValikVityaz**
fixes #341

### CODE REVIEW

File: sc-memory/sc-core/sc-store/sc_storage.c [hunk 1]
```diff
@@ -478,7 +478,9 @@ sc_result sc_storage_element_free(sc_memory_context *ctx, sc_addr addr)
         sum.len = SC_CHECKSUM_LEN;
       }
 
-      STORAGE_CHECK_CALL(sc_fs_storage_remove_content_addr(addr, &sum));
+      if (sc_element_is_checksum_empty(el) == SC_FALSE) {
```
Comment (deniskoronchik) [inline sc-memory/sc-core/sc-store/sc_storage.c:481]:
```suggestion
      if (sc_element_is_checksum_empty(el) == SC_FALSE) 
      {
```

File: sc-memory/tests/sc-memory/common/sc-link.cpp [hunk 2]
```diff
@@ -95,3 +95,19 @@ TEST_F(ScLinkTest, as_string)
   EXPECT_TRUE(link.Set<uint64_t>(7600000000));
   EXPECT_EQ(link.GetAsString(), "7600000000");
 }
+
+TEST_F(ScLinkTest, erase) {
```
Comment (deniskoronchik) [inline sc-memory/tests/sc-memory/common/sc-link.cpp:99]:
```suggestion
TEST_F(ScLinkTest, erase)
{
```

**Review by deniskoronchik — CHANGES_REQUESTED** [review]
codestyle

**Comment by deniskoronchik** [comment]
Update changelog

File: sc-memory/tests/sc-memory/common/sc-link.cpp [hunk 3]
```diff
@@ -95,3 +95,19 @@ TEST_F(ScLinkTest, as_string)
...
+  ScMemoryContext ctx(sc_access_lvl_make_min, "sc_link_erase");
```
Comment (deniskoronchik) [inline sc-memory/tests/sc-memory/common/sc-link.cpp:100]: Use `m_ctx` instead

**Review by KovalM — APPROVED** [review]

**Review by deniskoronchik — APPROVED** [review]

**Review by MikhailSadovsky — APPROVED** [review]

**Pull request merged**